xapian-letor/ranker/ranker.cc

```diff
@@ -192,32 +192,29 @@ write_to_file(const std::vector<Xapian::FeatureVector> & list_fvecs, const strin
     }
 }
 
+// Query file is in the format: <qid> '<query_string>'
 static std::pair<string, string>
 parse_query_string(const string & query_line, int line_number)
 {
-    vector<string> token;
-    size_t j = 0;
-    while (j < query_line.size()) {
-	size_t i = query_line.find_first_not_of(' ', j);
-	if (i == string::npos) break;
-	j = query_line.find_first_of(' ', i);
-	token.push_back(query_line.substr(i, j - i));
+    string::size_type i = 0;
+    string::size_type j = query_line.find_first_of(' ', i);
+    if (j == string::npos) {
+	throw LetorParseError("Could not parse Query file at line:" +
```

Comment:
I wonder if a more explicit message ("missing space between fields in Query file" or something) might be helpful here.

Reply:
fixed in a fixup commit.